### PR TITLE
Refactor NewXXXPlugin in framework_test to avoid package variables

### PR DESCRIPTION
**Pull request by hex108**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

/kind cleanup

> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Refactor NewXXXPlugin to avoid package variables to avoid race conditions. It is a follow-up for https://github.com/kubernetes/kubernetes/pull/81328.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

/sig scheduling

### CODE REVIEW

File: test/integration/scheduler/framework_test.go [hunk 1]
```diff
@@ -637,20 +650,21 @@ func TestNormalizeScorePlugin(t *testing.T) {
 		t.Errorf("Expected the pod to be scheduled. error: %v", err)
 	}
 
-	if scoreWithNormalizePlguin.numScoreCalled == 0 {
+	if scoreWithNormalizePlugin.numScoreCalled == 0 {
```
Comment (draveness) [inline test/integration/scheduler/framework_test.go]: good catch

**Review by draveness — APPROVED** [review]
/lgtm
/assign @Huang-Wei 
/priority important-soon

Thanks @hex108

File: test/integration/scheduler/framework_test.go [hunk 2]
```diff
@@ -126,11 +126,11 @@ var _ = framework.PostbindPlugin(&PostbindPlugin{})
 var _ = framework.UnreservePlugin(&UnreservePlugin{})
 var _ = framework.PermitPlugin(&PermitPlugin{})
 
-var scPlugin = &ScorePlugin{}
-
-// NewScorePlugin is the factory for score plugin.
-func NewScorePlugin(_ *runtime.Unknown, _ framework.FrameworkHandle) (framework.Plugin, error) {
-	return scPlugin, nil
+// NewScorePlugin returns a factory for score plugin with specified ScorePlugin.
+func NewScorePlugin(scorePlugin *ScorePlugin) framework.PluginFactory {
```
Comment (ahg-g) [inline test/integration/scheduler/framework_test.go]: do we really need to export those New* functions? should be enough if we do newScorePlugin etc.

File: test/integration/scheduler/framework_test.go [hunk 3]
```diff
@@ -563,6 +564,12 @@ func TestPrefilterPlugin(t *testing.T) {
 
 // TestScorePlugin tests invocation of score plugins.
 func TestScorePlugin(t *testing.T) {
+	// Create a plugin registry for testing. Register only a score plugin.
+	scPlugin := &ScorePlugin{}
```
Comment (ahg-g) [inline test/integration/scheduler/framework_test.go]: not your fault, but I dislike those short plugin names "scPlugin" etc. not descriptive and confusing, perhaps a followup cleanup PR to rename all of them properly, like s/scPlugin/scorePlugin, s/perPlugin/permitPlugin etc?

**Review by ahg-g — COMMENTED** [review]
Thanks @hex108 for a very good catch on the flake test!

File: test/integration/scheduler/framework_test.go (hunk 2 above)
Comment (hex108) [reply test/integration/scheduler/framework_test.go]: Agree :) Addressed it.

File: test/integration/scheduler/framework_test.go (hunk 3 above)
Comment (hex108) [reply test/integration/scheduler/framework_test.go]: Also addressed in this PR. PTAL

**Review by draveness — COMMENTED** [review]
looks good, could you please squash the commits

File: test/integration/scheduler/framework_test.go [hunk 4]
```diff
@@ -126,11 +126,11 @@ var _ = framework.PostbindPlugin(&PostbindPlugin{})
 var _ = framework.UnreservePlugin(&UnreservePlugin{})
 var _ = framework.PermitPlugin(&PermitPlugin{})
 
-var scPlugin = &ScorePlugin{}
-
-// NewScorePlugin is the factory for score plugin.
-func NewScorePlugin(_ *runtime.Unknown, _ framework.FrameworkHandle) (framework.Plugin, error) {
-	return scPlugin, nil
+// newScorePlugin returns a factory for score plugin with specified ScorePlugin.
+func newScorePlugin(scorePlugin *ScorePlugin) framework.PluginFactory {
```
Comment (Huang-Wei) [inline test/integration/scheduler/framework_test.go]: You can simply compose a generic `newPlugin`:

```go
func newPlugin(plugin framework.Plugin) framework.PluginFactory {
	return func(_ *runtime.Unknown, fh framework.FrameworkHandle) (framework.Plugin, error) {
		return plugin, nil
	}
}
```

And remove all `newXYZPlugin()` b/c all callers can just pass in specific plugin struct as a plugin interface obj.

Comment (hex108) [reply test/integration/scheduler/framework_test.go]: Addressed. If we need a more specific newXXXPlugin, we could add it later.

**Comment by Huang-Wei** [comment]
Thanks @hex108. Please squash the commits and then it'd good to /approve.

**Comment by hex108** [comment]
@Huang-Wei Addressed :) The failed test is not related.

**Comment by Huang-Wei** [comment]
/lgtm
/approve

Thanks, @hex108 !

**Comment by hex108** [comment]
@Huang-Wei Thanks! It seems there is something wrong with the /approve

**Pull request merged**